app/main.py

```diff
@@ -1,34 +1,105 @@
+        if (row, column) in cords_list:
+            index = cords_list.index((row, column))
+            return self.decks[index]
+
+    def fire(self, column: int, row: int) -> str:
+        deck = self.get_deck(row, column)
+        if deck:
+            deck.is_alive = False
+            deck.texture = "*"
+            if not self:
+                for deck in self.decks:
+                    deck.texture = "x"
+                self.is_drowned = True
+                return "Sunk!"
+            return "Hit!"
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def __bool__(self) -> bool:
+        return any(bool(deck) for deck in self.decks)
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def __len__(self) -> int:
+        return len(self.decks)
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(
+            self,
+            ships: list[tuple[tuple[int, int], tuple[int, int]]]
+    ) -> None:
+        self.ships: list[Ship] = self.generate_ships(ships)
+        self.field: dict[tuple[int, int], Ship] = self.generate_field()
+
+    @staticmethod
+    def generate_ships(
+            ships: list[tuple[tuple[int, int], tuple[int, int]]]
+    ) -> list[Ship]:
+        result = list()
+        for start, end in ships:
+            result.append(Ship(start, end))
+        return result
+
+    def generate_field(self) -> dict[tuple[int, int], Ship]:
+        field = dict()
+        for ship in self.ships:
+            cords_list = [(deck.column, deck.row) for deck in ship.decks]
+            for cords in cords_list:
+                field[cords] = ship
+        return field
+
+    def fire(self, location: tuple[int, int]) -> str:
+        if location in self.field:
+            return self.field[location].fire(*location)
+        return "Miss!"
+
+    def print_field(self) -> None:
+        surface = Surface(10, 10, "~")
+        for cords in self.field:
+            texture = self.field[cords].get_deck(*cords).texture
+            column, row = cords
+            surface.draw_point(row, column, texture)
+        print(surface)
+
+    def _validate_field(self) -> None:
+        assert len(self.ships) == 10
+        assert sorted(map(len, self.ships)) == [1, 1, 1, 1, 2, 2, 2, 3, 3, 4]
+        for cords in self.field:
```

Comment:
3 nested loops are not allowed when writing code, try to rewrite this method somehow